File: chardev 9/website/public/js/engine/entities/spell/SpellAuraOptions.js
/**
 * @constructor
 * @param {Array} serialized
 */
function SpellAuraOptions( serialized ) {
	this.stacks = serialized[0];
	this.procRate = serialized[1];
	this.procCharges = serialized[2];
}
SpellAuraOptions.prototype = {
		stacks: 0,
		procRate: 0,
		procCharges: 0
};

File: tools/cjt/src/org/chardev/cjt/entity/SpellAuraOptions.java
package org.chardev.cjt.entity;

public class SpellAuraOptions {
	protected final int stacks;
	protected final int procCharges;
	protected final int procRate;
	
	public SpellAuraOptions(int stacks, int procCharges, int procRate) {
		this.stacks = stacks;
		this.procCharges = procCharges;
		this.procRate = procRate;
	}
	
	public int getStacks() {
		return stacks;
	}
	
	public int getProcCharges() {
		return procCharges;
	}
	
	public int getProcRate() {
		return procRate;
	}
}
